languages/tree-sitter-stack-graphs-javascript/src/stack-graphs.tsg

```diff
@@ -3476,6 +3476,287 @@ inherit .return_or_yield
+  name:(_)@element_name)@jsx_opening_element {
+
+  node @jsx_opening_element.before_scope
+  node @jsx_opening_element.after_scope
+
+  edge @element_name.before_scope -> @jsx_opening_element.before_scope
+
+}
+
+(jsx_opening_element
+  name:(_)@element_name
+  !attribute)@jsx_opening_element
+{
+  
+  edge @jsx_opening_element.after_scope -> @element_name.after_scope
+
+}
+
+(jsx_opening_element
+  name:(_)@element_name
+  .
+  attribute:(_)@first_attr
+) {
+
+  edge @first_attr.before_scope -> @element_name.after_scope
+
+}
+
+(jsx_opening_element
+  attribute:(_)@left_attr
+  .
+  attribute:(_)@right_attr
+) {
+
+  edge @right_attr.before_scope -> @left_attr.after_scope
+
+}
+
+(jsx_opening_element
+  attribute:(_)@last_attr
+  .)@jsx_opening_element
+{
+
+  edge @jsx_opening_element.after_scope -> @last_attr.after_scope
+
+}
+
+(jsx_opening_element name:(identifier)@name) {
+
+  node @name.before_scope
+  node @name.after_scope
+
+  edge @name.after_scope -> @name.before_scope
+
+}
+
+(jsx_attribute)@jsx_attribute {
+
+  node @jsx_attribute.before_scope
+  node @jsx_attribute.after_scope
+
+  ; LATER-TODO scope propagation through empty child lists
+  ; currently unsupported by tree sitter queries
+  ; THIS IS A HUGE HACK AND MUST BE FIXED
+  edge @jsx_attribute.after_scope -> @jsx_attribute.before_scope
+
+}
+
+(jsx_attribute (_) . (_)@attr_value)@jsx_attribute {
+
+  edge @attr_value.before_scope -> @jsx_attribute.before_scope
+  edge @jsx_attribute.after_scope -> @attr_value.after_scope
+
+}
+
+(jsx_self_closing_element
+  name:(_)@element_name)@jsx_self_closing_element {
+
+  node @jsx_self_closing_element.before_scope
+  node @jsx_self_closing_element.after_scope
+  node @jsx_self_closing_element.value
+
+  edge @element_name.before_scope -> @jsx_self_closing_element.before_scope
+
+}
+
+(jsx_self_closing_element
+  name:(_)@element_name
+  !attribute)@jsx_self_closing_element
+{
+  
+  edge @jsx_self_closing_element.after_scope -> @element_name.after_scope
+
+}
+
+(jsx_self_closing_element
+  name:(_)@element_name
+  .
+  attribute:(_)@first_attr
+) {
+
+  edge @first_attr.before_scope -> @element_name.after_scope
+
+}
+
+(jsx_self_closing_element
+  attribute:(_)@left_attr
+  .
+  attribute:(_)@right_attr
+) {
+
+  edge @right_attr.before_scope -> @left_attr.after_scope
+
+}
+
+(jsx_self_closing_element
+  attribute:(_)@last_attr
+  .)@jsx_self_closing_element
+{
+
+  edge @jsx_self_closing_element.after_scope -> @last_attr.after_scope
+
+}
+
+(jsx_self_closing_element name:(identifier)@name) {
+
+  node @name.before_scope
+  node @name.after_scope
+
+  edge @name.after_scope -> @name.before_scope
+
+}
+
+(jsx_expression)@jsx_expression {
+
+  node @jsx_expression.before_scope
+  node @jsx_expression.after_scope
+
+  ; LATER-TODO scope propagation through empty child lists
+  ; currently unsupported by tree sitter queries
+  ; THIS IS A HUGE HACK AND MUST BE FIXED
+  edge @jsx_expression.after_scope -> @jsx_expression.before_scope
```

Comment:
Idem.

Reply:
I'm not sure I follow

Reply:
Oh! It's a single one instead of multiple! Gotcha

Reply:
Wait so for `is-empty` we write `if (is-empty @nodes) {` but for `none` we write `if none @node {`?? Curious.

Reply:
Yeah, the language design is a bit clunky there. I wanted to retain at least the possibility to type this at some point, and so I didn't want to use `(is-null ...)` to check for optional values.